stl/inc/chrono

```diff
@@ -5171,6 +5176,155 @@ namespace chrono {
+    template <class _Ty, class _CharT>
+    concept _Chrono_parse_spec_callbacks = requires(_Ty _At, basic_string_view<_CharT> _Sv, _Align _Aln) {
+        { _At._On_conversion_spec(_CharT{}, _CharT{}) } -> same_as<void>;
+        { _At._On_lit_char(_CharT{}) } -> same_as<void>;
+    };
+    // clang-format on
+
+    template <class _CharT>
+    struct _Chrono_specs {
+        _CharT _Lit_char = _CharT{0}; // any char other than {, }, %
+        char _Modifier   = '\0'; // either E or O
+        char _Type       = '\0';
+    };
+
+    template <class _CharT>
+    struct _Chrono_format_specs {
+        int _Width                   = 0;
+        int _Precision               = -1;
+        int _Dynamic_width_index     = -1;
+        int _Dynamic_precision_index = -1;
+        _Align _Alignment            = _Align::_None;
+        // At most one codepoint (so one char32_t or four utf-8 char8_t)
+        _CharT _Fill[4] = {' ', _CharT{0}, _CharT{0}, _CharT{0}};
+        // recursive definition in grammar, so could have any number of these with literal chars
+        vector<_Chrono_specs<_CharT>> _Chrono_specs_list;
+    };
+
+    // Model of _Chrono_parse_spec_callbacks that fills a _Chrono_format_specs with the parsed data
+    template <class _CharT>
+    class _Chrono_specs_setter {
+    public:
+        explicit constexpr _Chrono_specs_setter(_Chrono_format_specs<_CharT>& _Specs_) : _Specs(_Specs_) {}
+
+        // same as _Specs_setter
+        constexpr void _On_align(_Align _Aln) {
+            _Specs._Alignment = _Aln;
+        }
+
+        // same as _Specs_setter
+        constexpr void _On_fill(basic_string_view<_CharT> _Sv) {
+            if (_Sv.size() > 4) {
+                _THROW(format_error("Invalid fill (too long)."));
+            }
+
+            _STD fill(_Specs._Fill, _Specs._Fill + 4, _CharT{});
+            _STD copy(_Sv.begin(), _Sv.end(), _Specs._Fill);
+        }
+
+        constexpr void _On_width(int _Width) {
+            _Specs._Width = _Width;
+        }
+
+        constexpr void _On_precision(int _Prec) {
+            _Specs._Precision = _Prec;
+        }
+
+        constexpr void _On_conversion_spec(_CharT _Modifier, _CharT _Type) {
+            // NOTE: same performance note from _Basic_format_specs also applies here
+            const char _Char_mod  = static_cast<char>(_Modifier);
+            const char _Char_type = static_cast<char>(_Type);
+            if (_Char_mod != 'E' && _Char_mod != 'O') {
+                _THROW(format_error("Invalid modifier specification."));
+            }
+
+            if (_Type < 0 || _Type > (numeric_limits<signed char>::max)()) {
+                _THROW(format_error("Invalid type specification."));
+            }
+
+            _Chrono_specs<_CharT> _Conv_spec{_Char_mod, _Char_type};
+            _Specs._Chrono_specs_list.push_back(_Conv_spec);
+        }
+
+        constexpr void _On_lit_char(_CharT _Lit_ch) {
+            _Chrono_specs<_CharT> _Lit_char_spec{_Lit_ch};

```

Comment:
This initialization happens to be correct because `_Lit_char` is now the first data member. However, perhaps we should consistently use designated initializers with `_Chrono_specs`.